libs/vkd3d/command.c

```diff
@@ -1681,6 +1681,284 @@ static void d3d12_command_list_invalidate_current_pipeline(struct d3d12_command_
+                    1, &vk_clear_attachment, 1, &vk_clear_rect));
+        }
+    }
+}
+
+static void d3d12_command_list_clear_attachment_pass(struct d3d12_command_list *list, struct d3d12_resource *resource,
+        struct vkd3d_view *view, VkImageAspectFlags clear_aspects, const VkClearValue *clear_value, UINT rect_count,
+        const D3D12_RECT *rects)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+    VkAttachmentDescription attachment_desc;
+    VkAttachmentReference attachment_ref;
+    VkSubpassDependency dependencies[2];
+    VkSubpassDescription subpass_desc;
+    VkRenderPassBeginInfo begin_info;
+    VkRenderPassCreateInfo pass_info;
+    VkFramebuffer vk_framebuffer;
+    VkRenderPass vk_render_pass;
+    VkPipelineStageFlags stages;
+    VkAccessFlags access;
+    VkExtent3D extent;
+    VkResult vr;
+
+    attachment_desc.flags = 0;
+    attachment_desc.format = view->format->vk_format;
+    attachment_desc.samples = vk_samples_from_dxgi_sample_desc(&resource->desc.SampleDesc);
+    attachment_desc.loadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+    attachment_desc.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+    attachment_desc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+    attachment_desc.stencilStoreOp = VK_ATTACHMENT_STORE_OP_STORE;
+    attachment_desc.initialLayout = resource->common_layout;
+    attachment_desc.finalLayout = resource->common_layout;
+
+    attachment_ref.attachment = 0;
+    attachment_ref.layout = view->info.texture.vk_layout;
+
+    subpass_desc.flags = 0;
+    subpass_desc.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+    subpass_desc.inputAttachmentCount = 0;
+    subpass_desc.pInputAttachments = NULL;
+    subpass_desc.colorAttachmentCount = 0;
+    subpass_desc.pColorAttachments = NULL;
+    subpass_desc.pResolveAttachments = NULL;
+    subpass_desc.pDepthStencilAttachment = NULL;
+    subpass_desc.preserveAttachmentCount = 0;
+    subpass_desc.pPreserveAttachments = NULL;
+
+    if (!rect_count)
+    {
+        if (clear_aspects & (VK_IMAGE_ASPECT_COLOR_BIT | VK_IMAGE_ASPECT_DEPTH_BIT))
+            attachment_desc.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+
+        if (clear_aspects & (VK_IMAGE_ASPECT_STENCIL_BIT))
+            attachment_desc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+
+        /* Ignore 3D images as re-initializing those may cause us to
+         * discard the entire image, not just the layers to clear. */
+        if (clear_aspects == view->format->vk_aspect_mask &&
+                resource->desc.Dimension != D3D12_RESOURCE_DIMENSION_TEXTURE3D)
+            attachment_desc.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    }
+
+    if (clear_aspects & (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT))
+    {
+        stages = VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+        access = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
```

Comment:
Just using WRITE should be fine since CLEAR only accesses WRITE.

Reply:
Ok, we're not using CLEAR in all cases, so nvm this.

Reply:
I made this a bit more sophisticated, only uses `READ` if necessary now.